### DISCUSSION TITLE
Very slow download speed abraunegg-onedrive client using fedora 39.

### DISCUSSION THREAD

**Discussion by JuliGass** [abraunegg/onedrive#2672] [Q&A] [answered]
I've recently switched from kubuntu 23.10 to fedora 39 and installed the onedrive client as I did in my previous Ubuntu 22.04 and Kubuntu 23.10 installs, the client worked flawlessly both in Ubuntu 22.04 and Kubuntu 23.10, but now in Fedora the download speeds are awfully slow, what took ~3hs in Ubuntu/Kubuntu synchronizing now has taken +2 days in Fedora, even though they are pretty much the same files and size to download from my onedrive.

I don't know if I have messed up something in the install or if I have to config anything manually, I always used the default config in all the installs I have done. The only difference is the versions I have used: Ubuntu used 2.4.13, in Kubuntu used 2.4.23 and in Fedora used 2.4.25.

Any help would be greatly appreciated. Please ask if I should upload any additional information.

**Comment by abraunegg** [comment, marked as answer]
@JuliGass 
> installed the onedrive client as I did in my previous Ubuntu 22.04 and Kubuntu 23.10 installs, the client worked flawlessly both in Ubuntu 22.04 and Kubuntu 23.10
> ...
> The only difference is the versions I have used: Ubuntu used 2.4.13, in Kubuntu used 2.4.23 and in Fedora used 2.4.25.

Based on those versions you were using - you were installing from Ubuntu Universe and not the correct application as per https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-release-candidate-1/docs/ubuntu-package-install.md - so you were always running the incorrect and unsupported client version on Ubuntu.

> but now in Fedora the download speeds are awfully slow, what took ~3hs in Ubuntu/Kubuntu synchronizing now has taken +2 days in Fedora, even though they are pretty much the same files and size to download from my onedrive.

This will have nothing much to do with the client - but with your network card. The client, by default does not set any traffic download speed , but this can be done via https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-release-candidate-1/docs/usage.md#how-to-rate-limit-the-application-to-control-bandwidth-consumed-for-upload--download-operations

I would start by diagnosing your Fedora system and network speeds generally first, as this is where the issue most likely lives.

I would open up a discussion on Fedora forums to ask for help to diagnose why your network is potentially running slow.
